torchvision/datasets/imagenet.py

```diff
@@ -118,41 +108,97 @@ def extra_repr(self):
         return "Split: {split}".format(**self.__dict__)
 
 
-def parse_devkit(root):
-    idx_to_wnid, wnid_to_classes = parse_meta(root)
-    val_idcs = parse_val_groundtruth(root)
+def parse_devkit_archive(archive, meta_file=None):
+    """Parse the devkit archive of the ImageNet2012 classification dataset and save
+    the meta information in a binary file.
+
+    Args:
+        archive (str): Path to the devkit archive
+        meta_file (str, optional): Optional name for the meta information file
+    """
+    import scipy.io as sio
+
+    def parse_meta(devkit_root):
+        metafile = os.path.join(devkit_root, "data", "meta.mat")
+        meta = sio.loadmat(metafile, squeeze_me=True)['synsets']
+        nums_children = list(zip(*meta))[4]
+        meta = [meta[idx] for idx, num_children in enumerate(nums_children)
+                if num_children == 0]
+        idcs, wnids, classes = list(zip(*meta))[:3]
+        classes = [tuple(clss.split(', ')) for clss in classes]
+        idx_to_wnid = {idx: wnid for idx, wnid in zip(idcs, wnids)}
+        wnid_to_classes = {wnid: clss for wnid, clss in zip(wnids, classes)}
+        return idx_to_wnid, wnid_to_classes
+
+    def parse_val_groundtruth(devkit_root):
+        file = os.path.join(devkit_root, "data",
+                            "ILSVRC2012_validation_ground_truth.txt")
+        with open(file, 'r') as txtfh:
+            val_idcs = txtfh.readlines()
+        return [int(val_idx) for val_idx in val_idcs]
+
+    if meta_file is None:
+        meta_file = os.path.join(os.path.dirname(archive), META_FILE_NAME)
+
+    tmpdir = tempfile.mkdtemp()
+    extract_archive(archive, tmpdir)
```

Comment:
Should I include a MD5 check here in light of #1456? For the normal operation this check is performed within `ImageNet.extract_archives()`.

Reply:
Yes, a md5 check would be great!